stl/inc/xutility

```diff
@@ -193,18 +193,30 @@ using _Algorithm_int_t = conditional_t<is_integral_v<_Ty>, _Ty, ptrdiff_t>;
 
 // ITERATOR STUFF (from <iterator>)
 // ITERATOR TAGS (from <iterator>)
-struct input_iterator_tag {};
+struct input_iterator_tag {
+    explicit input_iterator_tag() = default;
+};
 
-struct output_iterator_tag {};
+struct output_iterator_tag {
+    explicit output_iterator_tag() = default;
+};
 
-struct forward_iterator_tag : input_iterator_tag {};
+struct forward_iterator_tag : input_iterator_tag {
+    explicit forward_iterator_tag() = default;
+};
 
-struct bidirectional_iterator_tag : forward_iterator_tag {};
+struct bidirectional_iterator_tag : forward_iterator_tag {
+    explicit bidirectional_iterator_tag() = default;
+};
 
-struct random_access_iterator_tag : bidirectional_iterator_tag {};
+struct random_access_iterator_tag : bidirectional_iterator_tag {
+    explicit random_access_iterator_tag() = default;
+};
 
 #ifdef __cpp_lib_concepts
-struct contiguous_iterator_tag : random_access_iterator_tag {};
+struct contiguous_iterator_tag : random_access_iterator_tag {
+    explicit contiguous_iterator_tag() = default;
+};

```

Comment:
Unfortunately, the "explicit default constructor" improvement can't be extended to these Standard types, so please revert this group of changes.

The issue is that WG21-N4849 [std.iterator.tags]/1 (see https://eel.is/c++draft/std.iterator.tags#1 for HTML instead of PDF) specifies the Standard tag types, with implicit default constructors, so we have to match that observable behavior. In contrast, we can do basically whatever we want with internal non-Standard types like `_Priority_tag` below.

Reply:
Oh okay let's revert so !
thanks a lot  !! I didn't aware about that